sql/codegen_alps.go

```diff
@@ -255,13 +267,37 @@ func newALPSTrainFiller(pr *extendedSelect, db *DB, session *pb.Session, ds *tra
 		ExitOnSubmit:        exitOnSubmit}, nil
 }
 
-func newALPSPredictFiller(pr *extendedSelect) (*alpsFiller, error) {
-	return nil, fmt.Errorf("alps predict not supported")
+func newALPSPredictFiller(pr *extendedSelect, session *pb.Session) (*alpsFiller, error) {
+	var ossID, ossKey *expr
+	var ok bool
+	if ossID, ok = pr.predAttrs["OSS_ID"]; !ok {
+		return nil, fmt.Errorf("the ALPS Predict job should specify OSS_ID")
+	}
+	if ossKey, ok = pr.predAttrs["OSS_KEY"]; !ok {
+		return nil, fmt.Errorf("the ALPS Predict job should specify OSS_KEY")
+	}
+	modelDir := fmt.Sprintf("oss://arks-model/%s/%s.tar.gz", session.UserId, pr.predictClause.model)
+
+	return &alpsFiller{
+		IsTraining:         true,
```

Comment:
Why predict should set `IsTraining:         true,`?

Reply:
done.